### PR TITLE
Initial ramp fitting support.

### PR DESCRIPTION
**Pull request by PaulHuwe**
This PR is to add ramp fitting support.

Tests: Fails as it depends on [#38]

Docs: Unneeded

Ticket: [#36]

### CODE REVIEW

**Review by dmggh — COMMENTED** [review]
The ramp fitting yaml files look fine to me.

File: src/rad/resources/schemas/ramp-1.0.0.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      tag: tag:stsci.edu:asdf/core/ndarray-1.0.0
+      ndim: 3
+      datatype: float32
+    pixeldq:
+      title: 2-D data quality array for all planes
+      tag: tag:stsci.edu:asdf/core/ndarray-1.0.0
+      ndim: 2
+      datatype: uint32
+    groupdq:
+      title: 3-D data quality array for each plane
```
Comment (perrygreenfield) [inline src/rad/resources/schemas/ramp-1.0.0.yaml:25]: Something about this seems inconsistent. Perhaps: individual 2-D data quality arrays for each plane?

File: src/rad/resources/schemas/ramp-1.0.0.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      tag: tag:stsci.edu:asdf/core/ndarray-1.0.0
+      ndim: 3
+      datatype: uint8
+    err:
+      title: Error array
```
Comment (perrygreenfield) [inline src/rad/resources/schemas/ramp-1.0.0.yaml:30]: Where does this error array come from? And what in the ramp does it apply to. Might as well be specific here for documentation purposes.

File: src/rad/resources/schemas/rampfitoutput-1.0.0.yaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+title: Ramp fit output schema
+
+allOf:
+- type: object
+  properties:
+    meta:
+      allOf:
+        - $ref: common-1.0.0
+    slope:
```
Comment (perrygreenfield) [inline src/rad/resources/schemas/rampfitoutput-1.0.0.yaml:14]: The size of the 3d dimension is the maximum number of segments determined for all pixels?

File: src/rad/resources/schemas/rampfitoutput-1.0.0.yaml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+      ndim: 3
+      datatype: float32
+    crmag:
+      title: Approximate CR magnitudes
+      tag: tag:stsci.edu:asdf/core/ndarray-1.0.0
+      ndim: 3
+      datatype: float32
+  required: [meta, slope, sigslope, yint, sigyint, pedestal, weights, crmag]
+  propertyOrder: [meta, slope, sigslope, yint, sigyint, pedestal, weights, crmag]
+- $ref: variance-1.0.0
```
Comment (PaulHuwe) [inline src/rad/resources/schemas/rampfitoutput-1.0.0.yaml:51]: This should be undone, as the variance variables are segment-specific and not the same variance variable block that is in variance schema.

File: src/rad/resources/schemas/ramp-1.0.0.yaml (hunk 1 above)
Comment (PaulHuwe) [reply src/rad/resources/schemas/ramp-1.0.0.yaml:25]: Changed the descriptions.

File: src/rad/resources/schemas/ramp-1.0.0.yaml (hunk 2 above)
Comment (PaulHuwe) [reply src/rad/resources/schemas/ramp-1.0.0.yaml:30]: It is the square root of exposure-level combined variance. I have modified the description.

File: src/rad/resources/schemas/rampfitoutput-1.0.0.yaml (hunk 3 above)
Comment (PaulHuwe) [reply src/rad/resources/schemas/rampfitoutput-1.0.0.yaml:14]: Yes.

File: src/rad/resources/schemas/rampfitoutput-1.0.0.yaml (hunk 4 above)
Comment (PaulHuwe) [reply src/rad/resources/schemas/rampfitoutput-1.0.0.yaml:51]: Fixed.

**Comment by PaulHuwe** [comment]
PR comments addressed, but this is blocked by [#40] (group support).

**Comment by nden** [comment]
Is this really blocked?

https://github.com/spacetelescope/rad/pull/40/files#r644855824

And is the `group` schema necessary for Roman, at least for now? What is it used for?

**Review by nden — APPROVED** [review]
LGTM, a minor comment about the change log

**Pull request merged**